scalyr_agent/tests/copying_manager_test.py

```diff
@@ -245,6 +245,29 @@ def test_schedule_log_path_for_removal(self):
         matchers = self._manager.log_matchers
         self.assertEquals( 0, len(matchers) )
 
+    def test_schedule_pending_log_path_for_removal(self):
+        config = {}
+        self.create_copying_manager( config )
+        self.fake_scan()
+
+        path = os.path.join( self._log_dir, "newlog.log" )
+        self.append_log_lines( path, "line1\n" )
+
+        log_config = {
+            "path": path
+        }
+
+        self._manager.add_log_config('unittest', log_config)
+        self._manager.schedule_log_path_for_removal('unittest', path)
+
+        self.fake_scan()
```

Comment:
Is there a reason you need to call `fake_scan` three times?  If there is a reason, you should document it here.  If there isn't, and it works with just one scan, let's remove the other ones.

Reply:
It does not work with fewer scans... I tried to document it in the comments in the new commit.

Reply:
Ok, that looks good, thanks.